### PR TITLE
feat: use gomail for shoutrrr SMTP

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves: https://github.com/flanksource/incident-commander/issues/551

### CODE REVIEW

File: notification/shoutrrr.go [hunk 1]
```diff
@@ -91,6 +91,25 @@ func Send(ctx *api.Context, connectionName, shoutrrrURL, title, message string,
 	var params *types.Params
 	if properties != nil {
 		params = (*types.Params)(&allProps)
+	} else {
+		params = &types.Params{}
+	}
+
+	// NOTE: Until shoutrrr fixes the "UseHTML" props, we'll use the mailer package
+	if service == "smtp" {
+		parsedURL, err := url.Parse(shoutrrrURL)
```
Comment (moshloop) [inline notification/shoutrrr.go:100]: I think we need to handle the smtp user/pass/host/port specified in the url  - only smtp://system should default to the environment variables

File: notification/shoutrrr.go [hunk 2]
```diff
@@ -88,11 +89,32 @@ func Send(ctx *api.Context, connectionName, shoutrrrURL, title, message string,
...
 	injectTitleIntoProperties(service, title, allProps)
 
-	var params *types.Params
+	params := &types.Params{}
 	if properties != nil {
 		params = (*types.Params)(&allProps)
 	}
 
+	// NOTE: Until shoutrrr fixes the "UseHTML" props, we'll use the mailer package
+	if service == "smtp" {
```
Comment (moshloop) [inline notification/shoutrrr.go:98]: But now this doesn't work for smtp://system ?

Comment (adityathebe) [reply notification/shoutrrr.go:98]: It does. At this point in code, the url is never `smtp://system`. It's already prepopulated. https://github.com/flanksource/incident-commander/blob/cf47ca1327b40638ce8e569691c270d3d6364723/notification/shoutrrr.go#L54-L60

**Comment by adityathebe** [comment]
https://github.com/containrrr/shoutrrr/pull/388

**Pull request merged**